code/modules/mining/abandoned_crates.dm

```diff
@@ -287,3 +287,48 @@
 		if(25)
 			new /obj/item/weapon/spellbook/oneuse/smoke(src)
 
+
+//Spooky special loot
+
+/obj/item/device/wisp_lantern
+	name = "spooky lantern"
+	desc = "This lantern gives off no light, but is home to a friendly wisp."
+	icon = 'icons/obj/lighting.dmi'
+	icon_state = "lantern-blue"
+	var/obj/effect/wisp/wisp
+
+/obj/item/device/wisp_lantern/attack_self(mob/user)
+	if(!wisp)
+		user << "The wisp has gone missing!"
+		return
+	if(wisp.loc == src)
+		user << "You release the wisp. It begins to bob around your head."
+		wisp.orbit(user, 20)
+		user.sight |= SEE_MOBS
+		icon_state = "lantern"
+	else
+		if(wisp.orbiting)
+			var/atom/A = wisp.orbiting
+			if(istype(A, /mob/living))
+				var/mob/living/M = A
+				M.sight -= SEE_MOBS // This doesn't work
+				M << "The wisp has returned to it's latern. Your vision returns to normal." //This works
+			wisp.stop_orbit() //This works
+		wisp.loc = src //This works
+		user << "You return the wisp to the latern."
+		icon_state = "lantern-blue"//This doesn't work
```

Comment:
Okay so it

1. Starts blue

2. Click it inhand, release wisp. It stays blue.

3. Click inhand, recall wisp. It turns off.

4. Click inhand, release wisp. It stays off.